subprojects/jacoco/src/main/java/org/gradle/internal/jacoco/JacocoAgentJar.java

```diff
@@ -34,7 +34,7 @@
     private static final VersionNumber V_0_7_6_0 = VersionNumber.parse("0.7.6.0");
 
     private final Project project;
-    private FileCollection agentConf;
+    private final PropertyState<FileCollection> agentConf;
```

Comment:
This should stay as non-final `FileCollection`, as you can already create a lazy `FileCollection`, or be changed to `ConfigurableFileCollection` (without changing `FileCollection` or `ConfigurableFileCollection` to understand `Provider`), which is already basically a property state. We don't want to be using `Provider<FileCollection>` anywhere.